### DISCUSSION TITLE
Support of final redirected URL in the output

### DISCUSSION THREAD

**Discussion by HolyBugx** [projectdiscovery/httpx#291] [Q&A] [answered]
Hello PD,

Recently in the latest version of httpx the redirection system had some changes, one of the changes is that when we use `-follow-redirects` or `-follow-host-redirects` the final redirected URL is shown inside brackets.

Previously, only the final redirected URL was in the output, but this recent change makes it hard to pipe httpx to other tools, needs some more grepping on our side especially if the output is in `.txt`. 

If possible provide a flag to only print the final redirected URL (the one that is inside brackets).

Kind Regards,
HolyBugx

**Comment by ehsandeep** [comment]
@HolyBugx here is what changed:-

Previously when `-follow-redirects` or `-follow-host-redirects` is used, httpx follows the redirect and you see status/title and all related information of the final URL, but in the output, you still see the same input host, not the final URL.

Now we added support to also print the Final URL as additional information, the output of the probe is still same as before.

If you are looking to parse output, `-json` flag has all the information and you can parse any field of your interest, for example 

```
chaos -d hackerone.com | httpx -json | jq -r .url
```

**Comment by rivalsec** [comment, marked as answer]
subfinder -d hackerone.com | httpx -follow-redirects -json | jq -r 'if (.["final-url"]) then .["final-url"] else .url end'

**Reply by ldionmarcil** [reply]
not sure if the syntax changed since then, but `final-url` is now `final_url`. as such, the command would be
`jq -r 'if (.["final_url"]) then .["final_url"] else .url end'`
